tmt/steps/provision/__init__.py

```diff
@@ -993,6 +1008,8 @@ def get_data_class(cls) -> type[GuestData]:
     #: Guest topology hostname or IP address for guest/guest communication.
     topology_address: Optional[str] = None
 
+    guest_logs: list[GuestLog] = []
```

Comment:
1. it's already in `Guest` class, `guest_` prefix is superfluous. `Guest.guest_logs` vs `Guest.logs`
2. a a class-level mutable list, I would be worried that `self.guest_logs.append()` below would add guest-specific logs to one shared list. The list should probably be initialized in `__init__()` instead.